build_tools/bazel/iree_check_test.bzl

```diff
@@ -99,7 +105,8 @@ def iree_check_single_backend_test_suite(
     """
     tests = []
     for src in srcs:
-        test_name = "_".join([name, src])
+        base_src = remove_prefix(src, "//iree/test/e2e/xla_ops:")
+        test_name = "_".join([name, base_src])
```

Comment:
Ooof this makes this super specific to a few tests because you're trying to reference files one level up. Let's not do this. I have a suggestion for another way to exclude the metal tests on non-darwin platforms.

The tests will build fine without it, we just want to exclude them at runtime. We already do this for drivers on various platforms (or we did, when our CI didn't support vulkan). You can add `--test_tag_filters="-driver=metal"` to the `non_darwin` config and put these back with the other things. Sorry for coming in late and contradicting Scott's suggestion...

Reply:
So one annoying thing here is that `--test_tag_filters` doesn't accumulate like other bazel flags (https://github.com/bazelbuild/bazel/issues/7322). I think this is still likely to be a better way to do this. Unfortunately it means when manually specifying additional test_tag_filters folks will have to re-add the `-driver=metal`, but I think this flag isn't used very often anyway.